cooltools/lib/common.py

```diff
@@ -532,15 +532,14 @@ def wrapper(*args, **kwargs):
                 mymap = pool.imap
             else:
                 mymap = map
-
-            func(*args, **kwargs, map=mymap)
-
+            try:
+                result = func(*args, **kwargs, map=mymap)
+            finally:
+                if pool != None:
```

Comment:
this should guarantee the destruction of the pool (and killing children-processes) in case `func` crashes or doesn't have `map` as an argument - right ?

Reply:
[Docs](https://docs.python.org/3/library/multiprocessing.html#module-multiprocessing.pool) seem to say that calling `terminate()` is also needed:

<img width="852" alt="image" src="https://github.com/open2c/cooltools/assets/1270651/f86d6700-53d1-48e8-8e42-c0538672cc18">